### ISSUE TITLE
 Hash::FieldHash::fieldhash call fails

### ISSUE THREAD

**Issue by eserte** [net-amqp-rabbitmq/net-amqp-rabbitmq#169]
The test suite immediately bails out if Hash::FieldHash is installed:
```
"/opt/perl-5.8.9/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- RabbitMQ.bs blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.bs 644
PERL_DL_NONLAZY=1 "/opt/perl-5.8.9/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
Usage: Hash::FieldHash::fieldhash(hash, name= NULL, package= NULL) at /home/cpansand/.cpan/build/2019060321/Net-AMQP-RabbitMQ-2.40001-0/blib/lib/Net/AMQP/RabbitMQ.pm line 715.
Compilation failed in require at t/000_use_ok.t line 7.
BEGIN failed--compilation aborted at t/000_use_ok.t line 7.
Bailout called.  Further testing stopped:  Could not load all modules
FAILED--Further testing stopped: Could not load all modules
```

**Comment by manchicken** [comment]
Which Hash::Field version?

~ Michael D. Stemle, Jr.

**Comment by eserte** [comment]
The latest: GFUJI/Hash-FieldHash-0.15.tar.gz

Which is not installable in modern perl installations (5.26+) because of https://github.com/gfx/p5-Hash-FieldHash/issues/4 , so probably the issue is mostly visible in older perls.

**Comment by manchicken** [comment]
Do you have the log?

**Comment by markwellis** [comment]
http://www.cpantesters.org/cpan/report/d1c0df2c-8686-11e9-b671-bd331f24ea8f

**Comment by markwellis** [comment]
So it seems to be related to https://github.com/net-amqp-rabbitmq/net-amqp-rabbitmq/blob/master/RabbitMQ.pm#L21:L22

Calling 'use Hash::FieldHash qw(fieldhash);' works fine, but doesn't work in the context of that if.

Should be an easy enough fix, I'll try and make some time to look at it tomorrow

**Comment by markwellis** [comment]
Fixed in #171
